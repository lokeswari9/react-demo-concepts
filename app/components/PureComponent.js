import React,{PureComponent} from 'react';

class PureComponents extends PureComponent{
    constructor(){
        super();
        this.state={
            age:5
        }
    }
    changeEvent=()=>{
        console.log('age is changed')
        this.setState({age:21});
    }
    render(){
        return(
            <div className="purecomp">
                <div>{this.state.age}</div>
                <button onClick={this.changeEvent}>click</button>
            </div>
        )
    }
}
export default PureComponents;