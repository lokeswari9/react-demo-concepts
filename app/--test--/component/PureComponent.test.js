import React from 'react';
import PureComponent from '../../components/PureComponent';
//import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
//import Adapter from 'enzyme-adapter-react-15';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

// describe('when pure component is given',()=>{
//     it('should render',()=>{
//         const wrapper = renderer.create(<PureComponent/>);
//         let tree=wrapper.toJSON();
//         expect(tree).toMatchSnapshot();
//     })
//     it('should render',()=>{
//         const wrapper = renderer.create(<PureComponent/>);
//         wrapper.toJSON().children[1].props.onClick();
//         expect(wrapper.toJSON()).toMatchSnapshot();
//     })
// })
describe('when pure component is given',()=>{
    it('should render',()=>{
        const wrapper = shallow(<PureComponent/>);
        expect(wrapper).toHaveLength(1);
    })
})