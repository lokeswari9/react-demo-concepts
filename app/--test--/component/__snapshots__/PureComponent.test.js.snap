// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when pure component is given should render 1`] = `
<div
  className="purecomp"
>
  <div>
    5
  </div>
  <button
    onClick={[Function]}
  >
    click
  </button>
</div>
`;

exports[`when pure component is given should render 2`] = `
<div
  className="purecomp"
>
  <div>
    21
  </div>
  <button
    onClick={[Function]}
  >
    click
  </button>
</div>
`;
